packages/mobx-state-tree/src/types/complex-types/model.ts

```diff
@@ -191,7 +191,7 @@ export class ModelType<S, T> extends ComplexType<S, T> implements IModelType<S,
 
     props<SP, TP>(
         properties: { [K in keyof TP]: IType<any, TP[K]> } & { [K in keyof SP]: IType<SP[K], any> }
-    ): IModelType<S & SP, T & TP> {
+    ): IModelType<S & SP & Snapshot<SP>, T & TP> {
```

Comment:
Solution from #635

Reply:
https://github.com/runningskull/mobx-state-tree/blob/0812fa2771659ef1e8819040d3b78266a130733c/packages/mobx-state-tree/src/types/complex-types/model.ts#L454-L463

I have no idea why,
But uncommenting the additional props declaration also fix this issue

Reply:
(done in latest commit)

Reply:
@mweststrate wdys?

Reply:
I have the impression that property types are broken completely with TS 2.7, are you guys experiencing the same? I mean even `x: types.number`  get's infered to `x: any` on the IModelType. Maybe we should just wait for TS 2.8 where we can fix this properly? cc @mattiamanzati

Reply:
I have the impression that property types are broken completely with TS 2.7, are you guys experiencing the same? I mean even `x: types.number`  get's infered to `x: any` on the IModelType. Maybe we should just wait for TS 2.8 where we can fix this properly? cc @mattiamanzati

Reply:
@mweststrate 
check this out
http://www.typescriptlang.org/play/#src=type%20IModelProperties%3CT%3E%20%3D%20%7B%20%5BK%20in%20keyof%20T%5D%3A%20T%5BK%5D%20%7D%0D%0Afunction%20inferer%3CT%3E(properties%3A%20IModelProperties%3CT%3E)%3A%20IModelProperties%3CT%3E%20%7B%0D%0A%20%20%20%20return%20properties%0D%0A%7D%0D%0A%0D%0A%2F%2F%20if%20you%20hover%20a%20you%20will%20see%20that%20prop1%20and%20prop2%20are%20inferred%20as%20any%0D%0Aconst%20a%20%3D%20inferer(%7B%0D%0A%20%20%20%20prop1%3A%20%22asd%22%2C%0D%0A%20%20%20%20prop2%3A%201%0D%0A%7D)%3B%0D%0A%0D%0Atype%20IMGOODTYPE%20%3D%20IModelProperties%3C%7B%20a%3A%20string%20%7D%3E%3B%0D%0A%2F%2F%20but%20here%2C%20the%20original%20type%20of%20prop1%20and%20prop2%20are%20restored%20%0D%0Atype%20imgoodTYPEEevenahasbadone%20%3D%20typeof%20a%3B

```ts
type IModelProperties<T> = { [K in keyof T]: T[K] };
function inferer<T>(properties: IModelProperties<T>): IModelProperties<T> {
  return properties;
}

// if you hover a you will see that prop1 and prop2 are:
// inferred as any on typescript 2.7.1
// inferred as string and number on typescript 2.6.2
const a = inferer({
  prop1: "asd",
  prop2: 1,
});

type IMGOODTYPE = IModelProperties<{ a: string }>;
// but here, the original type of prop1 and prop2 are restored,
// even in typescript 2.7.1
type imgoodTYPEEevenahasbadone = typeof a;
```